ludwig/backend/ray.py

```diff
@@ -889,59 +886,69 @@ def supports_multiprocessing(self):
     def read_binary_files(
         self, column: Series, map_fn: Optional[Callable] = None, file_size: Optional[int] = None
     ) -> Series:
-        column = column.fillna(np.nan).replace([np.nan], [None])  # normalize NaNs to None
-
-        # Assume that the list of filenames is small enough to fit in memory. Should be true unless there
-        # are literally billions of filenames.
-        # TODO(travis): determine if there is a performance penalty to passing in individual files instead of
-        #  a directory. If so, we can do some preprocessing to determine if it makes sense to read the full directory
-        #  then filter out files as a postprocessing step (depending on the ratio of included to excluded files in
-        #  the directory). Based on a preliminary look at how Ray handles directory expansion to files, it looks like
-        #  there should not be any difference between providing a directory versus a list of files.
+        # normalize NaNs to None
+        column = column.fillna(np.nan).replace([np.nan], [None])
+
         pd_column = self.df_engine.compute(column)
         fnames = pd_column.values.tolist()
-        idxs = pd_column.index.tolist()
 
         # Sample a filename to extract the filesystem info
         sample_fname = fnames[0]
         if isinstance(sample_fname, str):
-            fs, _ = get_fs_and_path(sample_fname)
-
-            read_datasource_fn_kwargs = {
-                "path_and_idxs": list(zip(fnames, idxs)),
-                "filesystem": PyFileSystem(FSSpecHandler(fs)),
-            }
-            if self.df_engine.partitioned and file_size is not None:
-                # Heuristic to determine parallelism: if the average file size is known (in bytes), then we can
-                # extrapolate to determine the total file size. We aim to have ~50MB partitions (5e7 bytes), so we
-                # set parallelism to be the total size / 50MB.
-                total_size = file_size * len(fnames)
-                parallelism = int(total_size / 5e7)
-                # Only set parallelism if it matches or exceeds the Ray default kwarg for parallelism
-                read_datasource_fn_kwargs["parallelism"] = max(RAY_DEFAULT_PARALLELISM, parallelism)
-
-            # The resulting column is named "value", which is a dict with two keys: "idx" and "data".
-            ds = ray.data.read_datasource(BinaryIgnoreNoneTypeDatasource(), **read_datasource_fn_kwargs)
-            df = self.df_engine.from_ray_dataset(ds)
-            # Persist the dataframe to prevent re-reading binary files on each subsequent map_objects call
-            df = self.df_engine.persist(df)
-            df["idx"] = self.df_engine.map_objects(df["value"], lambda row: int(row["idx"]))
-            df["value"] = self.df_engine.map_objects(df["value"], lambda row: row["data"])
-            df = df.rename(columns={"value": column.name})
+            import daft
```

Comment:
might want to put this in a try/except block telling the user to install the `distributed` extra

Reply:
Nice! @jppgks totally agree, do you want to push up a change with the try-except block?